consensus/safety-rules/src/error.rs

```diff
@@ -1,77 +1,41 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use consensus_types::common::Round;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 #[derive(Debug, Deserialize, Error, PartialEq, Serialize)]
 /// Different reasons for proposal rejection
 pub enum Error {
-    #[error("Timeout round, {0}, is incompatible with last votedx round, {1}")]
-    BadTimeoutLastVotedRound(u64, u64),
-
-    #[error("Timeout round, {0}, is incompatible with preferred round, {1}")]
-    BadTimeoutPreferredRound(u64, u64),
-
     #[error("Provided epoch, {0}, does not match expected epoch, {1}")]
-    IncorrectEpoch(u64, u64),
-
-    #[error("Internal error: {:?}", error)]
-    InternalError { error: String },
-
-    #[error("Invalid proposal: {}", {0})]
-    InvalidProposal(String),
-
-    #[error("Unable to verify that the new tree extneds the parent: {:?}", error)]
-    InvalidAccumulatorExtension { error: String },
-
+    BadEpoch(u64, u64),
```

Comment:
nit: "Bad" is often an unhelpful adjective as it can mean many different things.. Perhaps make these error names a little more descriptive/specific? e.g., (using the description text for each error), you could refactor:

BadEpoch -> UnexpectedEpoch
BadLastVotedRound -> IncompatibleLastVotedRound
BadPreferredRound -> IncompatiblePreferredRound